### PR TITLE
add signature issues to verification flow

### PR DESCRIPTION
**Pull request by PatoBeltran**
Clients that consume the API need a way of getting more information about the signature results.

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/Signing/SignatureIssue.cs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        public static SignatureIssue SignatureInformationUnavailableWarning(string message)
+        {
+            return new SignatureIssue(false, NuGetLogCode.NU3502, message);
+        }
+
+        public override string ToString()
+        {
+            var issueType = Fatal ? "ERROR" : "WARNING";
+
+            return $"{issueType}: NU{Code} {Message}";
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Packaging/Signing/SignatureIssue.cs]: I would let the logger put these together, this should have a ToLogMessage() method that returns an ILogMessage with the individual fields

File: src/NuGet.Core/NuGet.Packaging/Signing/SignatureIssue.cs [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        public NuGetLogCode Code { get; }
+
+        private SignatureIssue(bool fatal, NuGetLogCode code, string message)
+        {
+            Fatal = fatal;
+            Code = code;
+            Message = message;
+        }
+
+        public static SignatureIssue InvalidInputError(string message)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.Packaging/Signing/SignatureIssue.cs:24]: Add a verb prefix maybe

Comment (PatoBeltran) [reply src/NuGet.Core/NuGet.Packaging/Signing/SignatureIssue.cs:24]: @emgarten I was following the same standard as [CompatibilityIssue](https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.Commands/RestoreCommand/CompatibilityIssue.cs#L65). Should I prepend a `Create` to all of these static methods?

**Comment by PatoBeltran** [comment]
@emgarten 🔔

**Review by emgarten — APPROVED** [review]

**Pull request merged**